test/hcl.go

```diff
@@ -195,25 +196,22 @@ func hclFromAttributes(t *testing.T, attrs ast.Attributes) string {
 	attrList := attrs.SortedList()
 
 	filesRead := map[string][]byte{}
-	readFile := func(filename string) []byte {
+	readFileRange := func(filename string, frange hhcl.Range) []byte {
 		t.Helper()
 
 		if file, ok := filesRead[filename]; ok {
-			return file
+			return file[frange.Start.Byte:frange.End.Byte]
 		}
 
 		file, err := os.ReadFile(filename)
 		assert.NoError(t, err, "reading origin file")
 
 		filesRead[filename] = file
-		return file
+		return file[frange.Start.Byte:frange.End.Byte]
 	}
 
 	for _, attr := range attrList {
-		tokens, err := eval.GetExpressionTokens(
-			readFile(attr.Origin),
-			attr.Expr,
-		)
+		tokens, err := eval.TokensForExpressionBytes(readFileRange(attr.Origin, attr.Range()))
```

Comment:
Everytime I remember I did this...I cry inside :laughing:

Reply:
haha